test/google_search_data_viewer/keywords/keyword_test.exs

```diff
@@ -40,4 +40,45 @@ defmodule GoogleSearchDataViewer.Keywords.KeywordTest do
       assert Keyword.get_keyword_uploads_for_user(user2) == []
     end
   end
+
+  describe "get_keyword_upload/1" do
+    test "with an existing uploaded keyword, returns the keyword upload for the given id" do
+      user = insert(:user)
+      keyword_upload = insert(:keyword_upload, name: "dog", user: user)
+
+      assert keyword_upload ==
+               keyword_upload.id |> Keyword.get_keyword_upload() |> Repo.preload(:user)
+    end
+
+    test "with a non-existing uploaded keyword, returns nil" do
+      user = insert(:user)
+      insert(:keyword_upload, name: "dog", user: user)
+
+      assert Keyword.get_keyword_upload(-1) == nil
+    end
+  end
+
+  describe "update_keyword_upload_status/2" do
+    test "given an existing uploaded keyword and a new status of inprogress, updates the status to inprogress" do
+      user = insert(:user)
+      keyword_upload = insert(:keyword_upload, name: "dog", user: user)
```

Comment:
How about defining the status here? to ensure the test function would change the status

Reply:
The initial default value of `status` will be `pending`, but I can assert it's `pending` prior to changing it to `inprogress` if that is what you mean?

Reply:
```suggestion
      keyword_upload = insert(:keyword_upload, name: "dog", user: user, status: :pending)
```

Micky mean we define explicitly the status name in the `insert` method, then we can remove the assert status below
```
   assert keyword_upload.status == :pending
```